spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -391,6 +391,8 @@ public String toString() {
 
 		private boolean taskSchedulerExplicitlySet;
 
+		private boolean consumerPaused;
```

Comment:
I think this should be `volatile`. Even if that looks like we manipulate with this variable only in the `run()`, but when we call `stop()/start()` on the container, we lose that thread context and may run in some other one.

Reply:
It's not necessary; this is only ever accessed in `run()` by one thread (the consumer thread) - so there can't be any problems with cached values. `AMLC.paused` is volatile, though (and has to be).

Reply:
Yeah... Agreed. Now I see this in the `doStart()`:
```
this.listenerConsumer = new ListenerConsumer(this.listener, listenerType);
```

So, everything is perfect - merging...